### PR TITLE
Update AWS SDK to 1.12.7

### PR DESCRIPTION
**Pull request by justinsb**
```release-note
Update AWS SDK to 1.12.7
```

This includes support for all the latest goodies - in particular NLB and tagging as part of CreateVolume

cc @micahhausler  @gnufied 

Issue #54032

### CODE REVIEW

**Comment by justinsb** [comment]
Closing & reopening to trigger kubeadm e2e

**Comment by brendandburns** [comment]
Rubber stamp lgtm
/lgtm

**Comment by gnufied** [comment]
looks good to me, but we will need to find a high level approver.

**Comment by justinsb** [comment]
/retest

"boskos does not have a free gce-project at the moment"

**Comment by justinsb** [comment]
/retest

crash in etcd

File: build/root/BUILD.root [hunk 1]
```diff
@@ -65,6 +65,16 @@ filegroup(
...
+        "//kubernetes/cluster:all-srcs",
+        "//kubernetes/docs:all-srcs",
+        "//kubernetes/examples:all-srcs",
+        "//kubernetes/federation/cluster:all-srcs",
+        "//kubernetes/federation/develop:all-srcs",
+        "//kubernetes/hack:all-srcs",
+        "//kubernetes/test/e2e/testing-manifests:all-srcs",
+        "//kubernetes/test/images:all-srcs",
+        "//kubernetes/test/kubemark:all-srcs",
+        "//kubernetes/third_party/htpasswd:all-srcs",
```
Comment (BenTheElder) [inline build/root/BUILD.root]: I think maybe these aren't supposed to be here or update-bazel needs to be run, bazel-test will fail with these here and no build files

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/53561/pull-kubernetes-bazel-test/11273/

**Comment by justinsb** [comment]
Hooray!  `hack/update-bazel` was confused by some extra directories that had been created by a previous build.

**Comment by justinsb** [comment]
/retest

etcd again

**Comment by BenTheElder** [comment]
`k8s.io/kubernetes/cmd/kube-apiserver/app/testing TestCRD` stinks 😕 
/test pull-kubernetes-unit

**Comment by BenTheElder** [comment]
`TestCRD` x2 ❄️ 😕 
/retest

**Comment by BenTheElder** [comment]
@justinsb not sure about the verify-godeps issue, but at least unit is green now 🤷‍♂️

**Comment by BenTheElder** [comment]
see: https://github.com/kubernetes/kubernetes/pull/54714

**Comment by justinsb** [comment]
/test pull-kubernetes-e2e-kops-aws

job test flakes

**Comment by justinsb** [comment]
/test pull-kubernetes-e2e-kops-aws

More random flakes

**Pull request merged**